src/api-functions/open-ai-request/index.ts

```diff
@@ -14,11 +14,25 @@ import {
   SongType,
 } from "~/types";
 
+function checkIfChunkedContentExceedsLimit(chunkedTextContent: Array<string>) {
+  const totalWordsOfArticle = chunkedTextContent.flatMap((value) => value.split(" ")).length;
+  const max_word_length = 5000;
+  if (totalWordsOfArticle > max_word_length) {
+    const error = new Error(
+      `Error content too long, content exceeds ${max_word_length} words, content length: ${totalWordsOfArticle}`,
+    );
+    error.name = "insufficient length";
```

Comment:
I like how you separated this out from `getValidProps`.  We could perhaps use something like "Length error" or "Length constraint violation" here? And we can also use this same name  for insufficient length error (currently on Line 47)